golang/pre-publish/action.yml

```
@@ -11,13 +11,17 @@ inputs:
 runs:
   using: composite
   steps:
+    - name: Set up environment
+      shell: bash
+      run:
+         VERSION="${{ inputs.version }}" bash ${{ github.action_path }}/set-environment.sh
```

Nit: consider passing the version through the `env` option. Alternatively, pass it as an arg to `set-environment.sh` to make it more explicit.

```suggestion
      run:
         bash ${{ github.action_path }}/set-environment.sh
       env:
         VERSION: "${{ inputs.version }}"
```